backend/fms_core/viewsets/dataset.py

```diff
@@ -62,38 +62,24 @@ def add_run_processing(self, request, *args, **kwargs):
     @action(detail=True, methods=["patch"])
     def set_release_status(self, request, pk):
         data = request.data
-        release_status = data.get("release_status")
-        exceptions = data.get("exceptions")
-        filters = data.get("filters")
+        readset_updates: dict[int, ReleaseStatus.RELEASED | ReleaseStatus.BLOCKED] = data.get("updates", {})
+        filters = data.get("filters", {})
+
+        readset_ids = list(readset_updates.keys())
+        readsets = Readset.objects.filter(dataset=pk, id__in=readset_ids, **filters)
 
-        filtered_readsets = Readset.objects.filter(dataset=pk).exclude(validation_status=ValidationStatus.AVAILABLE)
-        if not filtered_readsets.exists():
-            transaction.set_rollback(True)
-            return HttpResponseBadRequest(f"Run must first be validated before release status can be changed.")
-        if filters:
-            filtered_readsets = filtered_readsets.filter(**filters)
         try:
-            # set release flag of all files except exceptions
-            included_readsets = filtered_readsets.filter(~Q(id__in=exceptions))
-            for included_readset in included_readsets:
-                included_readset.release_status = release_status
-                included_readset.release_status_timestamp = timezone.now()
-                included_readset.released_by = request.user
-                included_readset.save()
-
-            # set release flag of exceptions to the opposite flag
-            excluded_readsets = Readset.objects.filter(id__in=exceptions)
-            opposite_status = [ReleaseStatus.AVAILABLE, ReleaseStatus.BLOCKED, ReleaseStatus.RELEASED][release_status]
-            for excluded_readset in excluded_readsets:
-                excluded_readset.release_status = opposite_status
-                excluded_readset.release_status_timestamp = timezone.now()
-                excluded_readset.released_by = request.user
-                excluded_readset.save()
-        except Exception as err:
+            release_status_timestamp = timezone.now()
+            for readset in readsets:
+                readset.release_status = readset_updates[str(readset.id)]
+                readset.release_status_timestamp = release_status_timestamp
+                readset.released_by = request.user
+                readset.save()
+        except Exception as e:
             transaction.set_rollback(True)
-            return HttpResponseServerError(err)
+            return HttpResponseServerError(f"Error updating release status: {e}")
 
-        return Response(self.get_serializer(Dataset.objects.get(pk=pk)).data)
+        return Response(status=204)
```

Comment:
you removed the return of the data that was updated? what is the reasoning?

Reply:
The error messages are not caught in the front end (the operation fails silently in the console).

Reply:
I just felt it was pointless to respond with a dataset if the frontend is never using it anyway.
I should look into catching the error.

Reply:
Well you should at least tell that there was an error ... if you remove the line to set released_by it does fail; it just does not show anything on the frontend side.

Reply:
This was taken from a best practices for restful api's blog post: 
Updates & creation should return a resource representation
A PUT, POST or PATCH call may make modifications to fields of the underlying resource that weren't part of the provided parameters (for example: created_at or updated_at timestamps). To prevent an API consumer from having to hit the API again for an updated representation, have the API return the updated (or created) representation as part of the response.

In case of a POST that resulted in a creation, use a [HTTP 201 status code](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.5) and include a [Location header](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.30) that points to the URL of the new resource. Both of those in should be in addition to including the newly created resource representation as the body of the response.